src/core/render/index.js

```diff
@@ -41,58 +43,179 @@ function formatUpdated(html, updated, fn) {
+      //
+      // Full syntax examples:
+      // v-foo, v-foo[bar], v-foo-bar, v-foo:bar-baz.prop
+      //
+      // Shorthand syntax examples:
+      // @foo, @foo.bar, @foo.bar.baz, @[foo], :foo, :[foo]
+      //
+      // Markup examples:
+      // <div v-html>{{ html }}</div>
+      // <div v-text="msg"></div>
+      // <div v-bind:text-content.prop="text">
+      // <button v-on:click="doThis"></button>
+      // <button v-on:click.once="doThis"></button>
+      // <button v-on:[event]="doThis"></button>
+      // <button @click.stop.prevent="doThis">
+      // <a :href="url">
+      // <a :[key]="url">
+      const reHasDirective = /<[^>/]+\s([@:]|v-)[\w-:.[\]]+[=>\s]/;
+
+      vueMountData.push(
+        ...dom
+          .findAll('.markdown-section > *')
+          // Remove duplicates
+          .filter(elm => !vueMountData.some(([e, c]) => e === elm))
+          // Detect Vue content
+          .filter(elm => {
+            const isVueMount =
+              // is a component
+              elm.tagName.toLowerCase() in
+                (docsifyConfig.vueComponents || {}) ||
+              // has a component(s)
+              elm.querySelector(vueComponentNames.join(',') || null) ||
+              // has curly braces
+              reHasBraces.test(elm.outerHTML) ||
+              // has content directive
+              reHasDirective.test(elm.outerHTML);
+
+            return isVueMount;
+          })
+          .map(elm => {
+            // Clone global configuration
+            const vueConfig = Object.assign(
+              {},
+              docsifyConfig.vueGlobalOptions || {}
+            );
+
+            // Replace vueGlobalOptions data() return value with shared data object.
+            // This provides a global store for all Vue instances that receive
+            // vueGlobalOptions as their configuration.
+            if (vueGlobalData) {
+              vueConfig.data = function() {
+                return vueGlobalData;
+              };
+            }
+
+            return [elm, vueConfig];
+          })
+      );
+    }
 
-          app.mount(elm);
-        }
-      }
-      // Vue 2
-      else if (vueVersion === 2) {
-        const isAlreadyVue = Boolean(elm.__vue__ && elm.__vue__._isVue);
-
-        if (!isAlreadyVue) {
-          new window.Vue({
-            mounted: function() {
-              this.$destroy();
-            },
-          }).$mount(elm);
+    // Mount
+    for (const [mountElm, vueConfig] of vueMountData) {
+      const isVueAttr = 'data-isvue';
+      const isSkipElm =
+        // Is an invalid tag
+        mountElm.matches('pre, script') ||
+        // Is a mounted instance
+        isMountedVue(mountElm) ||
+        // Has mounted instance(s)
+        mountElm.querySelector(`[${isVueAttr}]`);
+
+      if (!isSkipElm) {
+        mountElm.setAttribute(isVueAttr, '');
+
+        if (vueVersion === 2) {
+          vueConfig.el = undefined;
+          new window.Vue(vueConfig).$mount(mountElm);
```

Comment:
What happens if user wants to do the following?

```html
<style>
  .example { border: 3px solid cornflowerblue }
</style>

<div class="example">
  Hello {{name}}, this content should have a blue border around it.
</div>
```

```js
window.$docsify = {
  vueMounts: {
    '.example': { data: () => ({ name: 'Amadar' }) }
  }
}
```

Will the content have a blue border?

Reply:
Here's how I solved that problem:

https://github.com/trusktr/trusktr.io/blob/54664bac7a3cb3d10c0d36ad0e4d6a11ab8e610e/meteor-app/client/imports/routes/index.js#L81-L91

Reply:
Works fine as-is:

```js
vueMounts: {
  '#example': {
    data() {
      return {
        message: 'Looks Good!',
      };
    },
  },
},
```

```html
<style>
  #example { border: 3px solid cornflowerblue }
</style>

<div id="example">
  Test: {{message}}
</div>
```

Screenshot:

<img width="468" alt="Screen Shot 2020-11-19 at 4 26 24 PM" src="https://user-images.githubusercontent.com/442527/99731780-0140b680-2a84-11eb-9858-4e041c042b1f.png">

Reply:
@jhildenbiddle Did you test with Vue 2 or Vue 3? Or both?